packages/@aws-cdk/integ-runner/test/cli.test.ts

```diff
@@ -51,11 +65,57 @@ describe('CLI', () => {
     ]]);
   });
 
+  test('find only TypeScript files', async () => {
+    await main(['--list', '--language', 'typescript', '--directory=test']);
+
+    expect(stdoutMock.mock.calls).toEqual([[
+      'language-tests/integ.typescript-test.ts\n',
+    ]]);
+  });
+
   test('can run with no tests detected', async () => {
     await main(['whatever.js', '--directory=test/test-data']);
 
     expect(stdoutMock.mock.calls).toEqual([]);
   });
+
+  test('app and test-regex override default presets', async () => {
+    await main([
+      '--list',
+      '--directory=test/test-data',
+      '--app="node {filePath}"',
+      '--test-regex="^xxxxx\.integ-test[12]\.js$"',
+    ]);
+
+    expect(stdoutMock.mock.calls).toEqual([[
+      [
+        'xxxxx.integ-test1.js',
+        'xxxxx.integ-test2.js',
+        '',
+      ].join('\n'),
+    ]]);
+  });
+
+
+  test('cannot use --test-regex by itself with more than one language preset', async () => {
+    await expect(() => main([
+      '--list',
+      '--directory=test/test-data',
+      '--language=javascript',
+      '--language=typescript',
+      '--test-regex="^xxxxx\.integ-test[12]\.js$"',
+    ])).rejects.toThrowError('Only a single "--language" can be used with "--test-regex". Alternatively provide both "--app" and "--test-regex" to fully customize the configuration.');
+  });
```

Comment:
Yes that's right. It's because we only have a single pair of `--app` and `--test-regex`. A possible future extension would be to allow configuring the presets or support passing in a map of `app -> test-regex`. 

In the meantime, if both is needed, users can always run `integ-runner` multiple times, e.g.:
```console
integ-runner --language typescript
integ-runner --language python --app="python3.2"
integ-runner --language go --test-regex=".*\.integ\.go"
```